### PR TITLE
Build payment method management

### LINKED ISSUE #3792: Tiny edit to plans-and-pricing page
Chris Johnson has requested this tiny removal of the word 'physcial'. This change should be made on mongoose. It is such a small edit that it seems unnecessary to update u.com as well. 
See Copydoc: https://docs.google.com/document/d/1R7FRZbfx_5YHFeZUD93HgT8RW4OCbbK4blOulB4D6EE/edit#

### PR DESCRIPTION
**Pull request by ClementChaumel**
## Done

- added the /advantage/payment-methods page
- added an endpoint for updating payment method

## QA

- go here https://ubuntu-com-9248.demos.haus/advantage/payment-methods?test_backend=true
- login
- Verify that it display your default payment method (if you don't have one, go [here](https://ubuntu-com-9248.demos.haus/advantage/subscribe?test_backend=true)  and make a purchase with a [dummy card](https://stripe.com/docs/testing) )
- Change it to another one (4242 4242 4242 424242.... or 5555 5555 5555 444444...)
- Verify that it updated correctly

## Issue / Card

Fixes #3792

## Screenshots

image
image

### CODE REVIEW

File: webapp/views.py [hunk 1]
```diff
@@ -984,6 +1051,54 @@ def post_anonymised_customer_info():
         return flask.jsonify({"error": "authentication required"}), 401
 
 
+def post_payment_method():
+    user_token = flask.session.get("authentication_token")
+    guest_token = flask.session.get("guest_authentication_token")
```
Comment (albertkol) [inline webapp/views.py:1056]: Is it possible to make this request a guest?

Comment (ClementChaumel) [reply webapp/views.py:1056]: No, if you go to the page without being logged in this is what you see
image

File: static/js/src/ua-payment-methods.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    })
+    .then((result) => {
+      setPaymentMethod(window.accountId, result.paymentMethod.id).then(() => {
+        location.reload();
+      });
+    })
+    .catch((error) => {
+      console.error(error);
+      this.classList.remove("is-processing");
+      this.innerHTML = "Confirm";
```
Comment (wgx) [inline static/js/src/ua-payment-methods.js:90]: @ClementChaumel "Confirm" is a bit non-specific, can we have "Update" instead?

**Comment by wgx** [comment]
@ClementChaumel copy change suggested: https://github.com/canonical-web-and-design/ubuntu.com/pull/9248/files#r578497964

File: static/js/src/ua-payment-methods.js (hunk 2 above)
Comment (ClementChaumel) [reply static/js/src/ua-payment-methods.js:90]: Sure!

**Comment by wgx** [comment]
👍🏽

File: webapp/views.py (hunk 1 above)
Comment (albertkol) [reply webapp/views.py:1056]: Allowing guests to make the post requests might not be necessary. Other than that it's looking good.

Comment (ClementChaumel) [reply webapp/views.py]: I removed it! :+1:

**Pull request merged**